elements/pfe-cta/src/pfe-cta.js

```diff
@@ -13,42 +13,120 @@ class PfeCta extends PFElement {
   }
 
   connectedCallback() {
     super.connectedCallback();
+
+    // Get the slot
     this._slot = this.shadowRoot.querySelector("slot");
+
+    // Attach the slotchange listener
     this._slot.addEventListener("slotchange", this._init);
+
+    if (this.children.length) {
+      this._init();
+    }
   }
 
   disconnectedCallback() {
+    // Remove the slot change listeners
     this._slot.removeEventListener("slotchange", this._init);
+
+    // Remove the focus state listeners
+    if(this.cta) {
+      this.cta.removeEventListener("focus", this._focusHandler);
+      this.cta.removeEventListener("blur", this._blurHandler);
+    }
   }
 
+  attributeChangedCallback(attr, oldValue, newValue) {
+    super.attributeChangedCallback(attr, oldValue, newValue);
+    // Strip the prefix form the attribute
+    attr = attr.replace("pfe-", "");
+    // If the observer is defined in the attribute properties
+    if (this[attr] && this[attr].observer) {
+      // Get the observer function
+      let observer = this[this[attr].observer].bind(this);
+      // If it's a function, allow it to run
+      if (typeof observer === "function") observer(attr, oldValue, newValue);
+    }
+  }
+
+  // Initialize the component
   _init() {
+    // Get the first child of the web component (light DOM)
     const firstChild = this.children[0];
+    const supportedTags = ["a", "button", "input"];
+    let supportedTag = false;
 
-    if (!firstChild) {
-      console.warn(
-        `${
-          PfeCta.tag
-        }:The first child in the light DOM must be an anchor tag (<a>)`
-      );
-    } else if (firstChild && firstChild.tagName.toLowerCase() !== "a") {
+    // If the first child does not exist or that child is not a supported tag
+    if (firstChild) {
+      supportedTags.forEach(tag => {
+        if (firstChild.tagName.toLowerCase() === tag) {
+          supportedTag = true;
+        }
+      });
+    }
+
+    if (!firstChild || !supportedTag) {
       console.warn(
         `${
           PfeCta.tag
-        }:The first child in the light DOM must be an anchor tag (<a>)`
+        }:The first child in the light DOM must be a supported call-to-action tag (<a>, <button>, <input>)`
       );
     } else {
-      this.link = this.querySelector("a");
+      // Capture the first child as the CTA element
+      this.cta = firstChild;
+
+      // Watch the light DOM link for focus and blur events
+      this.cta.addEventListener("focus", this._focusHandler);
+      this.cta.addEventListener("blur", this._blurHandler);
     }
   }
+
+  // On focus, add a class
+  _focusHandler(event) {
+    this.classList.add("focus-within");
+  }
+
+  // On focus out, remove that class
+  _blurHandler(event) {
+    this.classList.remove("focus-within");
+  }
+
+  _basicAttributeChanged(attr, oldValue, newValue) {
+    this[attr].value = newValue;
+  }
 }
 
 PFElement.create(PfeCta);
 
-export default PfeCta;
+export { PfeCta as default };
```

Comment:
Just curious, why this change?

Reply:
not sure; @kylebuch8 ?

Reply:
@KamiQuasi is requesting that we do this. And I've seen some posts on why this is better.

Reply:
That said, should we just leave this change out of this PR and do one big change where we update them all at the same time?

Reply:
removed

Reply:
Cool.  I'm not that familiar with arguments for the `export { Foo as default }` style.  I am ready to be swayed.  I just like the sweet simple syntax of `export default Foo`.